pkg/manager/config/config.go

```diff
@@ -0,0 +1,151 @@
+	MetricsAddr                            string
+	MetricsAccessFilter                    MetricsAccessFilter
+	ProbeAddr                              string
+	KongAdminURLs                          []string
+	KongAdminSvc                           OptionalNamespacedName
+	GatewayDiscoveryReadinessCheckInterval time.Duration
+	GatewayDiscoveryReadinessCheckTimeout  time.Duration
+	KongAdminSvcPortNames                  []string
+	ProxySyncSeconds                       float32
+	InitCacheSyncDuration                  time.Duration
+	ProxyTimeoutSeconds                    float32
+
+	// Kubernetes configurations
+	KubeconfigPath           string
+	IngressClassName         string
+	LeaderElectionNamespace  string
+	LeaderElectionID         string
+	LeaderElectionForce      string
+	Concurrency              int
+	FilterTags               []string
+	WatchNamespaces          []string
+	GatewayAPIControllerName string
+	Impersonate              string
+	EmitKubernetesEvents     bool
+	ClusterDomain            string
+
+	// Ingress status
+	PublishServiceUDP       OptionalNamespacedName
+	PublishService          OptionalNamespacedName
+	PublishStatusAddress    []string
+	PublishStatusAddressUDP []string
+
+	UpdateStatus                bool
+	UpdateStatusQueueBufferSize int
+
+	// Kubernetes API toggling
+	IngressNetV1Enabled           bool
+	IngressClassNetV1Enabled      bool
+	IngressClassParametersEnabled bool
+	UDPIngressEnabled             bool
+	TCPIngressEnabled             bool
+	KongIngressEnabled            bool
+	KongClusterPluginEnabled      bool
+	KongPluginEnabled             bool
+	KongConsumerEnabled           bool
+	ServiceEnabled                bool
+	KongUpstreamPolicyEnabled     bool
+	KongServiceFacadeEnabled      bool
+	KongVaultEnabled              bool
+	KongLicenseEnabled            bool
+	KongCustomEntityEnabled       bool
+
+	// Gateway API toggling.
+	GatewayAPIGatewayController        bool
+	GatewayAPIHTTPRouteController      bool
+	GatewayAPIReferenceGrantController bool
+	GatewayAPIGRPCRouteController      bool
+
+	// GatewayToReconcile specifies the Gateway to be reconciled.
+	GatewayToReconcile OptionalNamespacedName
+
+	// SecretLabelSelector specifies the label which will be used to limit the ingestion of secrets. Only those that have this label set to "true" will be ingested.
+	SecretLabelSelector string
+
+	// ConfigMapLabelSelector specifies the label which will be used to limit the ingestion of configmaps. Only those that have this label set to "true" will be ingested.
+	ConfigMapLabelSelector string
+
+	// Admission Webhook server config
+	AdmissionServer AdmissionServerConfig
+
+	// Diagnostics and performance
+	EnableProfiling                 bool
+	EnableConfigDumps               bool
+	DumpSensitiveConfig             bool
+	DiagnosticServerPort            int
+	DisableRunningDiagnosticsServer bool // TODO(czeslavo): instead of this toggle, move the server out of the internal.Manager
```

Comment:
Do we have an issue for this that we could link here?

Reply:
Created an issue and linked to it in the comment: https://github.com/Kong/kubernetes-ingress-controller/issues/7285